### DISCUSSION TITLE
Water vapor stratospheric bias

### DISCUSSION THREAD

**Discussion by cecilehannay** [NCAR/amwg_dev#27] [TF1: Unified physics for L58/L93]
This thread is about the water vapor stratospheric bias and attempt to fix it.

**Comment by cecilehannay** [comment]
On Dec 6, 2021.

Hi All,

Just following up with some action items from this morning's mtg. The mtg is getting a bit large and kind of hard to stay on topic (which is my fault!). So I'm just sending this out to a few of us.

1. submit a run w/ additional diffusion in clubb. If this doesn't give good results, we are going to go "nuclear" as Julio says; add a constant background diffusion, possibly for a select region. For the clubb diffusion experiment, Cecile you can just copy these source mods that turn it on?
/glade/u/home/aherring/src/cam6_3_035.tphysac/usr_src/clubbdiffuse/model_flags.F90

2. Re-submit some runs w/ the full moisture budget (coding in progress; I'll send Cecile mods when they are done). Then I can do some analysis on how moisture is getting injected into the stratosphere, and if any of the budget terms explain the drying up there. I would like to do this for L32, L48, L58 and L58+zm2. I believe we have these runs setup so they are using the same initial conditions. Budget terms are usually pretty robust using just one year of data. Isla showed that we need to run for three years to get moisture equilibrated. So what does everyone think, run for 4 years? Or is that overkill?

3. Piecewise linear vertical remapping. I think this is a good idea because that piecewise constant run is out to lunch; it's a whole new climate. So I'm not convinced that the q changes we see are the increased implicit diffusion signal. Peter, would you mind putting together these mods for Cecile?

And I forget to show results at the mtg, but using the latest cam code base (+ new physics reordering and a proper ZM2 implementation) doesn't look so bad (light blue). I think we should use this code base going forward for all experiments.

Other than that, I'll try to do my analysis for the wet season only since that's where the action's at.

@adamrher

**Comment by cecilehannay** [comment]
Hi All,

Following up w/ a summary and action items from today's mtg. I've assigned names to the action items but pls don't take them as orders; especially w/ AGU some of us may not get to them in time for next week's mtg.

I am pleased that we agreed on a reasonable target tape recorder based on the MLS obs product. My feeling is that our L58+zm2 runs aren't as far off as I once thought (I had thought our target was FV L70, but that tape recorder is perhaps too moist compared w/ MLS).

I would ask that if you'd rather skip over the actions items below, that if you can, to read the last section of this email. There, I am attempting to establish a consensus on what we do know based on expts over the last few months.

Action items:

(1) re-run L58+zm2 using new code base, but w/ clubb's trapezoidal rules off. This is to try to revert our runs w/ the new code base to the drier solutions in the older tag. Cecile, this just entails:
./xmlchange CAM_CONFIG_OPTS='-phys cam_dev' --append
clubb_l_trapezoidal_rule_zm = .true.
clubb_l_trapezoidal_rule_zt = .true.

(2) Run w/ a change in ZM2 parcel perturbation to try and recover the older ZM2 solution (Rich).

(3) run amwg diagnostics on the tau=7200s run versus default tau run, and vs obs. Increasing the convective time-scale helps alleviate the dry bias, but I suspect we are just introducing an error to counteract another error. So let's see how realistic this run is. (me or Cecile)

(4) Phase two of explicit vertical diffusion expts, move to a constant background diffusion to target UTLS. Code mods to be provided by Julio.

(5) Tape Recorder to do's (Isla). Add MLS observations, as well as the L48 (zm1) run, additional clubb diffusion run and tau=7200s run. 

(6) More analysis of q the budget using map plots (me), to try to characterize the ventilation process in time/space.

-----
I have put together a single slide that focuses on the L32, L48, L58 and L58+zm2 runs to establish what I hope are two not too controversial points:

(1) Dry bias arises from increasing resolution in the free troposphere / lower stratosphere
(2) Dry bias is NOT due to sensitivity of ZM to vert. res

Do we agree? Feel free to opine. Apologies if the slide is a bit dense; I tried to explain everything on it. One more point that we had established a while back is that raising the top of the model (L58->L93) does not change the dry bias. So it's not a top down issue.

Thanks
@adamrher 

![drybias_status](https://user-images.githubusercontent.com/9723220/146078291-91939eda-cf69-4319-bfb6-21fc999bb76f.png)

[drybias_status.pdf](https://github.com/NCAR/amwg_dev/files/7714680/drybias_status.pdf)

**Reply by islasimpson** [reply]
I updated three of the plots to include MLS.  

[https://www.cgd.ucar.edu/staff/islas/watervapor/Q50_vs_Q300.pdf](https://www.cgd.ucar.edu/staff/islas/watervapor/Q50_vs_Q300.pdf)
[https://www.cgd.ucar.edu/staff/islas/watervapor/taperecorder.pdf](https://www.cgd.ucar.edu/staff/islas/watervapor/taperecorder.pdf)
[https://www.cgd.ucar.edu/staff/islas/watervapor/Q_90hpa_seasonalcycle.pdf](https://www.cgd.ucar.edu/staff/islas/watervapor/Q_90hpa_seasonalcycle.pdf)


![Screen Shot 2021-12-15 at 10 37 46 AM](https://user-images.githubusercontent.com/9723220/146237147-642ca340-99f4-48f1-9d5c-eb8bd0f451d0.png)
![Screen Shot 2021-12-15 at 10 36 45 AM](https://user-images.githubusercontent.com/9723220/146237153-36436ee0-7c4d-4c04-ab34-f437eb59f113.png)
![Screen Shot 2021-12-15 at 10 35 40 AM](https://user-images.githubusercontent.com/9723220/146237158-58382f77-a78e-4d27-8bef-65922e970fca.png)

Bottom line:

(1) Climatological annual average water vapor in MLS is closest to dz1000 in the stratosphere although dz1000 is slightly moister..  It is more moist in the troposphere but I'm not sure to what extent we trust it there?  The lowest level in the dataset is ~315hPa and I'm plotting 300hPa so it's at the edge of the data.  But I don't know enough about MLS to know whether it's trust worthy there.

(2) dz1000 isn't perfect though when considering the seasonality, the dry phase is a big dry and the wet phase a bit wet.

**Reply by adamrher** [reply]
Thanks Isla! I don't understand why those links are dead though ... i assume you probably have use the hyperlink button.

**Reply by cecilehannay** [reply]
I included the plots in the post.

**Comment by islasimpson** [comment]
I did use the hyperlink button.


On Wed, Dec 15, 2021 at 10:31 AM Adam Herrington ***@***.***>
wrote:

> Thanks Isla! I don't understand why those links are dead though ... i
> assume you probably have use the hyperlink button.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/amwg_dev/discussions/27#discussioncomment-1814938>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJ7HQPYMSP5B7VWJCDZVS73URDGGXANCNFSM5KB4PBNQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Isla Simpson
Scientist 2, Climate Analysis Section
Climate and Global Dynamics Laboratory
National Center for Atmospheric Research
P.O. Box 3000 Boulder, CO 80307
Tel:303-497-1763
http://www.cgd.ucar.edu/staff/islas/

**Comment by cecilehannay** [comment]
Update on actions items:

(1) re-run L58+zm2 using new code base, but w/ clubb's trapezoidal rules on. This is to try to revert our runs w/ the new code base to the drier solutions in the older tag. 
clubb_l_trapezoidal_rule_zm = .true.
clubb_l_trapezoidal_rule_zt = .true.

_**=> This is [f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2_trapT.001.hf2](https://github.com/NCAR/amwg_dev/issues/28)
@cecilehannay:  currently running**_

(2) Run w/ a change in ZM2 parcel perturbation to try and recover the older ZM2 solution. 

_**@swrneale: is it easy for you to provide that code**_ 

(3) run amwg diagnostics on the tau=7200s run versus default tau run, and vs obs. Increasing the convective time-scale helps alleviate the dry bias, but I suspect we are just introducing an error to counteract another error. So let's see how realistic this run is. 

_**@cecilehannay:  Diags done: [diags](https://webext.cgd.ucar.edu/FWscHIST/f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2_tau7200.001.hf/atm/)**_

(4) Phase two of explicit vertical diffusion expts, move to a constant background diffusion to target UTLS. Code mods to be provided by Julio.

_**@JulioTBacmeister: will you be able to provide that code.**_  

(5) Tape Recorder to do's. Add MLS observations, as well as the L48 (zm1) run, additional clubb diffusion run and tau=7200s run.

_**@islasimpson: done ! (see plot above)**_
 
(6) More analysis of q the budget using map plots  to try to characterize the ventilation process in time/space.

_**@adamrher will update**_

**Reply by islasimpson** [reply]
I guess (5) is not totally done as I haven't added the new runs.  Will you be posting on here when these have reached 10 years?

**Reply by adamrher** [reply]
those runs are done and in /glade/scratch/hannay/archive/ ... There are now two options though ... new and old code base; Id suggest plotting the new code base runs since they are moister:

f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2
f.e21.FWscHIST.ne30_L48_cam6_3_035.tphysac_reorder.001.hf2
f.e21.FWscHIST.ne30_L48_B10_cam6_3_035.tphysac_reorder.001.hf2
f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2.001.hf2

And the two expts referenced in (5):
f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2_tau7200.001.hf
f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_zm2_reorder_clubbdiffusion.001.hf2

that is all!

**Reply by cecilehannay** [reply]
The latest run with w/ clubb's trapezoidal rules on is completed. 

All the runs done with the new code (moister) are in: 
/glade/scratch/hannay/archive/ 

Run sequence L32->L58/zm2
[f.e21.FWscHIST.ne30_L32_cam6_3_035.tphysac_reorder.001.hf2](https://github.com/NCAR/amwg_dev/issues/16)
[f.e21.FWscHIST.ne30_L48_cam6_3_035.tphysac_reorder.001.hf2](https://github.com/NCAR/amwg_dev/issues/18)
[f.e21.FWscHIST.ne30_L48_B10_cam6_3_035.tphysac_reorder.001.hf2](https://github.com/NCAR/amwg_dev/issues/20)
[f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2.001.hf2](https://github.com/NCAR/amwg_dev/issues/19)

Sensitivity study: 
[f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2_tau7200.001.hf](https://github.com/NCAR/amwg_dev/issues/25)
[f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_zm2_reorder_clubbdiffusion.001.hf2](https://github.com/NCAR/amwg_dev/issues/15)
[f.e21.FWscHIST.ne30_L48_BL10_cam6_3_035.tphysac_reorder_zm2_trapT.001.hf2](https://github.com/NCAR/amwg_dev/issues/28)

@islasimpson: if you want to recreate your plot.

**Reply by islasimpson** [reply]
Thanks.  Are there any runs that we don't care about that much anymore that I could remove from the plots so that it doesn't become overwhelming?

**Reply by swrneale** [reply]
Get rid of the KE run I did with the wrong settings #17.

**Reply by adamrher** [reply]
Id say get rid of the detmods run, but keep around the detmods+entr4.

@swrneale possible to get Cecile code mods to revert the parcel perturbation back to the ZM2 we were running with the older code base (is it tiedke_add?)?

I've renewed interest in that because the "revert clubb trapezoidal rules" run that just finished didn't recover the drier solution from the older code base.

**Reply by swrneale** [reply]
I'm checking on these now. It's not explicitly tiedke_add, but it could be argued to be similar since the parcel temp tpert= tparcel+teidke_add+0.5K. Basically the parcel temperature that was used in the previous zm2 runs was the temperature of the launch level, where that launch level was 0.5*pbl_height. The new runs use a parcel temperature that is backed out from a boundary layer mixed MSE and q. So I am presuming the parcel temperature is higher.

**Reply by islasimpson** [reply]
Updated figures...

[Tape recorder](https://www.cgd.ucar.edu/staff/islas/watervapor/taperecorder.pdf)
[Q 90hPa  seasonal cycle](https://www.cgd.ucar.edu/staff/islas/watervapor/Q_90hpa_seasonalcycle.pdf)
[50hPa Q versus 300hPa Q](https://www.cgd.ucar.edu/staff/islas/watervapor/Q50_vs_Q300.pdf)

All new sims are fitting with in the cloud of our other sims.  Best is the one without zm2. tau7200 has increased the water vapor a bit compared to its equivalent without the tau7200.  clubbdiffusion, and trapT don't look like they have changed stratospheric water vapor.

**Comment by JulioTBacmeister** [comment]
Thanks Isla. It is helpful to have the MLS water vapor on the plots.  We
need to figure out whether the no-ZM2 runs are better because the CFL
limits on ZM are being driven by a thinner cloud base. @Rich can we tweak
the limiter so that is about as active with ZM2 as without?

**Comment by adamrher** [comment]
> We need to figure out whether the no-ZM2 runs are better because the CFL
limits on ZM are being driven by a thinner cloud base. @rich can we tweak
the limiter so that is about as active with ZM2 as without?

One way to approximate the stricter CFL limits in an L32 ZM1 run is to change this line 

`        mumax(i) = max(mumax(i), mu(i,k)/dp(i,k))` <---divide ```dp``` by 2 or something so that it approximates the L58 grid.

I'm skeptical that the limiter is why the L58 ZM1 runs have weaker ZM behavior. The CFL limiter isn't preventing ZM2 from having juicier mass fluxes in L58, of which they are.

**Comment by swrneale** [comment]
I finally got to showing how the ZM2 changes are impact the ZM tendencies in the lowest level. The figure shows the lowest model level daily tendencies from a January of days for each simulation. Original L58 simulations have many more strong cooling events from ZM than in the courser L32. L48 looks very similar to L48; not surprising given the same surface level thickness. The original '019' tag and corrected '035' tag blunt the strong cooling tendencies to be less extreme than L32 even - maybe this could be controlled with a vertical length scale shallower than 0.5*PBLH?? Anyway both the zm2 versions look quite similar, so I should take a look to see what is causing the recent increase in column humidity. Oh and the threshold is PRECC>1mm/day to be counted.
![pdf_zm2_ZMDT](https://user-images.githubusercontent.com/8239194/146850779-ee65d217-09c3-4929-b439-507519c2b97a.png)

**Reply by adamrher** [reply]
I don't completely understand the processes governing the tendencies at the lowest model layer. Is the "subcloud layer" just one grid level in ZM1 and 0.5*PBL in ZM2? And so the 1/dz_subcld used in computing the tendencies is much larger in ZM1?

**Reply by adamrher** [reply]
Rich, just to be clear the moister solution using the "035" branch is not limited to ZM2 runs, the ZM1 runs ([turquoise and magenta](https://webext.cgd.ucar.edu/FSCAM/FWscHIST/L32_REF/12132021/temp_prof.Q.frac.newcode.pdf)) are also systematically moisture'er. But the "035" ZM2 flavor appears to have juicier mass fluxes, since detrained condensate is proportional to mass fluxes, and the detrainment is much larger. If the parcel properties in the "035" version are warmer/moisture (which I think is consistent with backing out T/q properties characteristic of an entire 0.5xPBL layer, as opposed to the "019" version where you just take the properties at the top of 0.5xPBL? Do I have that right?), then I would expect larger CAPE values in the "035" ZM2.

If this describes the ZM2 discrepancy, I don't think there's anything wrong with the "035" version;  it makes more sense to define the parcel properties the "035" way, IMO.

**Comment by adamrher** [comment]
Hi All,

I've been looking at Marine Sc in the MAGIC region in the runs, which refers to a 2013 campaign of ship borne measurements from container ships traveling from Long Beach to Honolulu. I'm plotting meridional averages over this [polygon in grey](https://webext.cgd.ucar.edu/FSCAM/cam6_3_006.dev_FSCAM_T42_T42.clubbmf/MAGIC_FORC_TO_CAM_JJA2010/temp_locations.pdf). The panel plot for most of our runs are [here](https://webext.cgd.ucar.edu/FSCAM/FWscHIST/L32_REF/01162022/temp_xztransect.MAGIC.CLOUD.pdf). There are two observational products for comparison. They are both Satellite based and so not the actual MAGIC campaign data.

My first impression is that L32 really struggles to resolve the inversion layer (specifically around 140W), as the grid coarsens significantly above 1500 m. This layer is better resolved in L58, but the clouds look more realistic in L32. I suspect this is due to a change in the large scale subsidence circulation above the inversion with enhanced vertical resolution L58 (this should be easy to check and I'll make a note to do so). ZM2 however brings back some of the Marine Sc. I suspect this is due to the more restrictive trigger, with ZM2 being less active in the MAGIC region. Updrafts in mass flux schemes like ZM tend to penetrate the inversion layer and mix free tropo air into the Sc cloud deck (i.e., cloud top entrainment) and destroy Sc clouds. So ZM2 seems to be a welcome improvement to the L58 grid in this respect (@swrneale).

The clubb-exp1 and exp2 indicate that we can tune our L58 clouds to be brighter, if we need them to be. However, these impact clouds globally so we will to more careful tuning when the time comes.

[edit - it does look like the [inversion layer is stronger](https://webext.cgd.ucar.edu/FSCAM/FWscHIST/L32_REF/01162022/temp_xztransect.MAGIC.STABILITY.pdf) at L58, although I should regrid everything to L32 to be sure.

**Comment by islasimpson** [comment]
Here is a plot assessing how our various CAM7 tests compare with other models in the CMIP6 archive in stratospheric water vapor.  I haven't put all the details on here about which run is which or which CMIP6 models you see.  But the point is that you can see CESM2 and CESM2-WACCM in the CMIP6 group and they were spot-on compared to MLS.  Our CAM7 tests have a dry bias, so are less perfect (as we already know), but we can see that it is a relatively small bias, compared to some other models in the CMIP6 archive.  Obviously, having models that are worse is not the target - it is better to be perfect.  But our stratospheric water vapor isn't embarrassing at this point.

![CMIP6_Q](https://user-images.githubusercontent.com/41842751/154766363-17cd113f-20a9-4115-9fb0-c61d2ccee91c.jpeg)

**Reply by adamrher** [reply]
awesome plot Isla! w.r.t the CAM7 tests, I'm wondering if we can have a different color for all the L58+zm2 runs, since we know the L32,L48,L58(zm1) runs are not real options for us.

**Reply by cecilehannay** [reply]
Really interesting plot.

**Comment by swrneale** [comment]
An idea what the error bars might be?

**Comment by islasimpson** [comment]
Here's an updated version addressing the comments of @adamrher and @swrneale.  I put all the L58+zm2 runs as filled circles and the others as open circles.  For the error bars - I should explain what I'm plotting.  For CMIP6 I'm using 1980-2000 climatologies.  You can see from the multiple members in some cases that the uncertainty range on 20y climatologies is small.  But for the CAM7 tests I'm using 1985-1990 (6 years, all with the same SSTs).  The red dotted lines show the range of 6 year climatologies from MLS (noting that there are only 16 years in this dataset).  The green shading shows the range of 1985-1990 climatologies across the 10 members of the CESM2 GOGA simulations.  So, errors are pretty small.

![CMIP6_Q](https://user-images.githubusercontent.com/41842751/154777669-5f4c2c36-f6ad-41b7-90f7-6dbfade555b3.jpeg)

**Comment by dan800** [comment]
UPDATE:

1. Doug Kinnison has experimented with adjusting the fall velocity to address the bias in the entry value. He made the two adjustments AG suggested:
+ micro_mg_dcs = 1e-3 (was 5e-4). "DCS is the size (in microns) threshold for conversion to snow. Basically it converts the ice mass at larger sized to snow and it falls out faster."
+ ! mass-weighted cloud ice fallspeed (m/s), vtrmi. (added the 0.2 reduction factor)
vtrmi = vtrmi * 0.2.

2. Constraining the upper boundary condition in L32 tests: see https://github.com/ESCOMP/CAM/issues/533#issuecomment-1084545694

Both seem to improve the dry stratosphere bias.
